### PR TITLE
Device software

### PR DESCRIPTION
**Pull request by apt105**
New page for critical device error codes in device software, and device encryption and portnums in device development section. https://meshtastic-54qfrqtsi-apt105.vercel.app/docs/software/device/critical-error-codes

### CODE REVIEW

**Comment by apt105** [comment]
@mc-hamster Do you want me to remove your email from the portnums page?

**Comment by sachaw** [comment]
I'd add an admonition at the top stating that this is derived from the protobufs and have a link to it.
Also, I think the emails can stay as they are the respective developers and POC for their plugins

File: website/docs/developers/device/encryption.md [hunk 1]
```diff
@@ -1,16 +1,15 @@
 ---
-id: crypto
-title: Encryption in Meshtastic
-sidebar_label: Crypto
+id: encryption
+title: Meshtastic encryption
+sidebar_label: Encryption
 ---
```
Comment (sachaw) [inline website/docs/developers/device/encryption.md:5]: The protobufs will need to be updated as they contain a link to this page.

File: website/docs/software/device/critical-faults.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+id: critical-error-codes
+title: Critical error codes
+sidebar_label: Critical error codes
+---
+
+The device might report these fault codes on the screen, but it will also be outputted on the device serial output. If you encounter a fault code, please post on the forum and we'll try to help.
+
+| Name | Number | Description |
+| ---- | ------ | ----------- |
+| None | 0 |  |
```
Comment (sachaw) [inline website/docs/software/device/critical-faults.md:11]: I'd remove this

**Review by sachaw — CHANGES_REQUESTED** [review]

**Comment by mc-hamster** [comment]
> @mc-hamster Do you want me to remove your email from the portnums page?

Yes please :)

**Comment by apt105** [comment]
I'll put a link to the associated GitHub accounts instead of the email addresses.

File: website/docs/developers/device/encryption.md (hunk 1 above)
Comment (apt105) [reply website/docs/developers/device/encryption.md:5]: PR done https://github.com/meshtastic/Meshtastic-protobufs/pull/20

File: website/docs/software/device/critical-faults.md (hunk 2 above)
Comment (apt105) [reply website/docs/software/device/critical-faults.md]: Removed the "none" line, or do you mean all of those rows?

**Comment by jfirwin** [comment]
I think the overview material could remain, and point a link to the protobufs section to replace the table or any duplication of materials?

I'll need to check when not on mobile.

File: website/docs/software/device/critical-faults.md (hunk 2 above)
Comment (apt105) [reply website/docs/software/device/critical-faults.md]: https://meshtastic-git-device-software-apt105.vercel.app/docs/software/device/critical-error-codes

**Comment by apt105** [comment]
Table removed and link to protobufs put in https://meshtastic-git-device-software-apt105.vercel.app/docs/developers/device/portnum

**Comment by jfirwin** [comment]
Should the bulletpoints be a table? I didn't catch it last time I looked, but there's two "fields" within each bulletpoint. What are your thoughts?

**Comment by apt105** [comment]
Yup, looks much better in a table, done!

**Comment by apt105** [comment]
I think all the requested changes so far have now been done.

**Comment by sachaw** [comment]
Please fix merge conflict

**Pull request merged**